Session_5/array_func.js

```diff
@@ -0,0 +1,9 @@
+// Array Helpers
+// http://www.codewars.com/kata/array-helpers
+
+Array.prototype.square = function() { return this.map(x => x * x) }
```

Comment:
you are combining es5 with es 6 

Array.prototype.square = function() { return this.map(x => x * x) } 

should be

Array.prototype.square = ()  => this.map(x => x * x)

Reply:
By using "Array.prototype.square = () => this.map(x => x * x)" I was getting this error as "this" in arrow functions and normal function are not same. Please guide. 
TypeError: this.map is not a function
    at Array.square
    at /home/codewarrior/index.js:21:30

One query do we have to use arrow function forcefully in ES6, means can't mix old function definition?

Reply:
Arrow functions are mandated in es6+.

In this case try this.prototype.map/ here this refers to the current context. you can manipulate accordingly